scale-encode/src/impls/mod.rs

```diff
@@ -48,64 +44,98 @@ use core::{
     ops::{Range, RangeInclusive},
     time::Duration,
 };
-use scale_info::{PortableRegistry, TypeDef, TypeDefPrimitive};
+use scale_type_resolver::{visitor, FieldIter, Primitive, ResolvedTypeVisitor, TypeResolver};
+
+// Useful to help encode key-value types or custom variant types manually.
+// Primarily used in the derive macro.
+pub use composite::{Composite, CompositeField};
+pub use variant::Variant;
+
+fn resolve_type_and_encode<
+    'resolver,
+    R: TypeResolver,
+    V: ResolvedTypeVisitor<'resolver, TypeId = R::TypeId, Value = Result<(), Error>>,
+>(
+    types: &'resolver R,
+    type_id: &R::TypeId,
+    visitor: V,
+) -> Result<(), Error> {
+    match types.resolve_type(type_id, visitor) {
```

Comment:
style: could be `map_err(|e| Error::new(ErrorKind::TypeResolvingError(e.to_string())))` instead of match

Reply:
Ah alas it'd be a bit more complicated than that because in the Ok branch we unwrap one level of results too

Reply:
yeah, doesn't really matter to me either is fine